lib/mongo/crypt/encryption_io.rb

```diff
@@ -158,6 +158,62 @@ def feed_kms(kms_context, tls_options)
         end
       end
 
+      # Adds a key_alt_name to the key_alt_names array of the key document
+      # in the key vault collection with the given id.
+      def add_key_alt_name(id, key_alt_name)
+        key_vault_collection.find_one_and_update(
+          { _id: id },
+          { '$addToSet' => { keyAltNames: key_alt_name } },
+        )
+      end
+
+      # Removes the key document with the given id
+      # from the key vault collection.
+      def delete_key(id)
+        key_vault_collection.delete_one(_id: id)
+      end
+
+      # Finds a single key document with the given id.
+      def get_key(id)
+        key_vault_collection.find(_id: id).first
+      end
+
+      # Returns a key document in the key vault collection with
+      # the given key_alt_name.
+      def get_key_by_alt_name(key_alt_name)
+        key_vault_collection.find(keyAltNames: key_alt_name).first
+      end
+
+      # Finds all documents in the key vault collection.
+      def get_keys
+        key_vault_collection.find.to_a
```

Comment:
I would prefer to call `.to_a` from the spec runner and here return the `Collection::View` which would 1) permit users to refine the query further, e.g. sort the keys, and 2) would maintain batched iteration behavior of find rather than always loading the entire set of keys into memory.